### DISCUSSION TITLE
使用TCP/UDP 动态代理, 想实现不限制client ip 或 server ip进行转发，求建议

### DISCUSSION THREAD

**Discussion by chenjinfeng0505** [apache/apisix#8124] [Q&A]
背景：使用 TCP/UDP动态代理实现请求转发功能，
规则中有：
server_addr: 接受 Stream Route 连接的 APISIX 服务器的地址。
server_port: 接受 Stream Route 连接的 APISIX 服务器的端口。
remote_addr: 发出请求的客户端地址。

实现目标： 将apisix收到的请求全部转发至指定地址。
问题： 客户端ip不固定，因此不能使用remote_addr来进行转发；
           因服务 server ip 会因网元重启发生变更，无法指定server  ip来进行转发，（ps:  每次重启后，重新根据新的service ip进行设置，但这种方式仍会造成部分请求的丢失）

请问有没有更好的方式可以不限制client ip 或 service ip 实现请求的转发？（可以限制端口）

**Comment by tzssangglass** [comment]
try
```
curl http://127.0.0.1:9180/apisix/admin/stream_routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "server_port": $your_server_port,
    "upstream": {
        "nodes": {
            "127.0.0.1:1995": 1
        },
        "type": "roundrobin"
    }
}'
```

**Reply by chenjinfeng0505** [reply]
Thanks.  I know.
I have another question.  During the request forwarding process,  the client ip remains unchanged.  Which parameter should I set.

**Reply by tzssangglass** [reply]
> I have another question. During the request forwarding process, the client ip remains unchanged. Which parameter should I set.

I did not understand

**Reply by chenjinfeng0505** [reply]
我的目的是通过动态代理将终端请求转发10.18.217.220。
目前我设置的规则是：
curl http://127.0.0.1:9180/apisix/admin/stream_routes/10 -H 'X-API-KEY: ef403f335dce4906a2962b8f015a6d2d' -X PUT -d '
{
    "server_port":80,
     "scheme":"http",
    "upstream": {
        "nodes": {
            "X.X.X.X:80": 1
        },
        "type": "roundrobin",
        "pass_host": "pass"
    }
}'
但是node服务收到的请求信息中client ip 为apisix服务器ip, 并不是终端请求的client ip。如果想要保持client ip不变，请求有什么建议吗？

**Reply by soulbird** [reply]
It looks like you want to do a transparent proxy with APISIX? The existing configuration of APISIX cannot be implemented yet. You need to know some knowledge about Nginx transparent proxy and iptables.

**Reply by chenjinfeng0505** [reply]
Ok，Thanks

**Reply by tokers** [reply]
> It looks like you want to do a transparent proxy with APISIX? The existing configuration of APISIX cannot be implemented yet. You need to know some knowledge about Nginx transparent proxy and iptables.

@Qiuqiu0505 Try the nginx directive proxy_bind. It may help you.
